PhysicsTools/PatAlgos/plugins/PATPackedCandidateProducer.cc

```diff
@@ -99,7 +103,9 @@ pat::PATPackedCandidateProducer::PATPackedCandidateProducer(const edm::Parameter
   PuppiCandsMap_(consumes<edm::ValueMap<reco::CandidatePtr> >(iConfig.getParameter<edm::InputTag>("PuppiSrc"))),
   PuppiCands_(consumes<std::vector< reco::PFCandidate > >(iConfig.getParameter<edm::InputTag>("PuppiSrc"))),
   PuppiCandsNoLep_(consumes<std::vector< reco::PFCandidate > >(iConfig.getParameter<edm::InputTag>("PuppiNoLepSrc"))),
-  minPtForTrackProperties_(iConfig.getParameter<double>("minPtForTrackProperties"))
+  minPtForTrackProperties_(iConfig.getParameter<double>("minPtForTrackProperties")),
+  usePuppi_(!iConfig.getParameter<edm::InputTag>("PuppiSrc").encode().empty() || 
+	    !iConfig.getParameter<edm::InputTag>("PuppiNoLepSrc").encode().empty())
```

Comment:
I would move the calls to `consumes<>` for the Puppi stuff inside the body of the method, inside a `if(usePuppi_) {...}`    (or bring the usePuppi_ higher up in the list of arguments, and do e.g. `PuppiWeight_(usePuppi_ ? consumes<...>() : edm::EDGetTokenT<...>())`)